react-js/styling-react.md

```diff
@@ -0,0 +1,109 @@
+
+- [Video: Styling of React App in 3 different ways](https://www.youtube.com/watch?v=j5P9FHiBVNo) - Using external CSS, internal and CSS modules.
+- [Video: Bootstrap in React](https://www.youtube.com/watch?v=wdbzLi0tWq0) - Bootstrap in React can be used using CDN or by installing the package. 
+
+### Lesson
+
+- [Slides: CSS in React ](https://drive.google.com/open?id=15oauTP4afibqMvNKW5XgVC1qraYUKxY-)
+
+### Guided Practice
+> #### Creating a simple div with InLine and External CSS and converting it into Bootstrap.
+
+1. Inline And External CSS
+  - App.js
+    - Creating a simple div using and add custom styling to it.
+    - ```
+      <div className="my-div">
+		<div style={{backgroundColor:"red"}}>Inner Div 1</div>
+		<div style={{backgroundColor:"blue"}}>Inner Div 2</div>
+		<div style={{backgroundColor:"green"}}>Inner Div 3</div>
+	  </div>
+      ```
+  - CSS 
+    - Make the *div* inside *my-div* class as flex box and add a little bit of padding.
+    - ```
+       .my-div
+	   {
+	     margin: 10px 0;
+	     display: flex;
+	     width: 100vw;
+	     background-color: darkorange;
+       }
+      .my-div div
+	  {
+	   flex-basis: 25%;
+	   max-width: 25%;
+	   padding: 10px;
+      }
+      ```
+2. Using Bootstrap
+  - App.js
+    - ```
+	    <div className="row mt-2 mb-2" style={{backgroundColor:"darkorange"}}>
+		<div className="col-3 p-2" style={{backgroundColor:"red"}}>Inner Div 1</div>
+		<div className="col-3 p-2" style={{backgroundColor:"blue"}}>Inner Div 2</div>
+		<div className="col-3 p-2" style={{backgroundColor:"green"}}>Inner Div 3</div>
+		</div>
+      ```
+
+
+### Common Mistakes & Misconceptions
+
+- Combining different classes based on certain state/props.
+  - When a component is receiving a prop and depending upon the value of that prop the CSS is rendered.
+  ```
+	<button className = {this.props.color ? "error" : "" + "btn-class" }>ButtonComp </button>
+  ```
+  - If color is received as prop by the component then **className** will be 
+    - > className = "error btn-class" 
+  - else
+    - > className = "btn-class"
+
+- Wrong path provided while importing CSS.
+  
+  - If the path provided for external CSS file or module in the JSX file is incorrect then react throws an error i.e it will not be able to build the webpack.
+
+- CamelCase is supported in ReactJS for CSS styles.
+	- using **class** instead of **className** gives warning.
+	- error encounters if any inline CSS property is used as **two words** like **background-color** because inline CSS is treated as an object in react.
+	- > Incorrect Way - ``` <button style={{background-color:"red" }}> ```
+	- > Correct Way   ``` <button style={{backgroundColor:"red"}} > ```
+
+	
+### Supplemental Materials
+- [React Strap](https://reactstrap.github.io/) - Provides predefined components for React styled with Bootstrap.
+  - Why to choose Reactstrap ?
+    - Directly use components instead of  bootstrap typing classes and passing attributes as props.
```

Comment:
Please put these lines in a syntax that uses “With Reactstrap, you can...” or “It allows you to...”